diff --git a/src/unix/config/discover.ml b/src/unix/config/discover.ml
index 7bc6d23..3362f0d 100644
--- a/src/unix/config/discover.ml
+++ b/src/unix/config/discover.ml
@@ -304,7 +304,7 @@ let exec_name = ref "a.out"
 let use_libev = ref true
 let use_pthread = ref true
 let use_unix = ref true
-let os_type = ref "Unix"
+let os_type = ref "Win32"
 let android_target = ref false
 let ccomp_type = ref "cc"
 let libev_default = ref true
@@ -568,7 +568,7 @@ let () =
     get exec_name "default_executable_name";
     get ccomp_type "ccomp_type";
     get system "system";
-    get os_type "os_type";
+    (* get os_type "os_type"; *)
     let get var name default =
       try
         let () =