diff --git a/testsuite/Makefile b/testsuite/Makefile
index 0a7325b..b89c68c 100644
--- a/testsuite/Makefile
+++ b/testsuite/Makefile
@@ -32,7 +32,7 @@ else # Windows
   ifeq "$(FLEXDLL_SUBMODULE_PRESENT)" ""
     FLEXLINK_PREFIX=
   else
-    ROOT:=$(shell cd .. && pwd| cygpath -m -f -)
+    ROOT:=$(shell cd .. && pwd)
     EMPTY=
     FLEXLINK_PREFIX:=OCAML_FLEXLINK="$(ROOT)/boot/ocamlrun \
                                      $(ROOT)/flexdll/flexlink.exe" $(EMPTY)
diff --git a/byterun/Makefile b/byterun/Makefile
index 9b92a58..d0d4aee 100644
--- a/byterun/Makefile
+++ b/byterun/Makefile
@@ -49,17 +49,7 @@ endif
 # On Windows, OCAML_STDLIB_DIR needs to be defined dynamically
 
 ifeq "$(UNIX_OR_WIN32)" "win32"
-# OCAML_STDLIB_DIR needs to arrive in dynlink.c as a string which both gcc and
-# msvc are willing parse without warning. This means we can't pass UTF-8
-# directly since, as far as I can tell, cl can cope, but the pre-processor
-# can't. So the string needs to be directly translated to L"" form. To do this,
-# we take advantage of the fact that Cygwin uses GNU libiconv which includes a
-# Java pseudo-encoding which translates any UTF-8 sequences to \uXXXX (and,
-# unlike the C99 pseudo-encoding, emits two surrogate values when needed, rather
-# than \UXXXXXXXX). The \u is then translated to \x in order to accommodate
-# pre-Visual Studio 2013 compilers where \x is a non-standard alias for \u.
-OCAML_STDLIB_DIR = $(shell echo $(LIBDIR)| iconv -t JAVA | sed -e 's/\\u/\\x/g')
-CFLAGS += -DOCAML_STDLIB_DIR='L"$(OCAML_STDLIB_DIR)"'
+CFLAGS += -DOCAML_STDLIB_DIR='L"$(LIBDIR)"'
 endif
 
 CFLAGS += $(IFLEXDIR)
