diff --git a/src/Makefile b/src/Makefile
index 97dc074..2dc3d1f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,7 +1,7 @@
-OCAMLC=ocamlc
-OCAMLOPT=ocamlopt
-OCAMLDEP=ocamldep
-OCAMLMKLIB=ocamlmklib
+OCAMLC=ocamlfind ocamlc
+OCAMLOPT=ocamlfind ocamlopt
+OCAMLDEP=ocamlfind ocamldep
+OCAMLMKLIB=ocamlfind ocamlmklib
 OCAMLFIND=ocamlfind
 INSTALL_DATA=install -m 644
 INSTALL_DLL=install
diff --git a/toplevel/Makefile b/toplevel/Makefile
index 8c91b0b..e77d0ba 100644
--- a/toplevel/Makefile
+++ b/toplevel/Makefile
@@ -1,5 +1,5 @@
-OCAMLC=ocamlc
-OCAMLDEP=ocamldep
+OCAMLC=ocamlfind ocamlc
+OCAMLDEP=ocamlfind ocamldep
 OCAMLFIND=ocamlfind
 
 CAMLCFLAGS=-I ../src -I +compiler-libs \
