diff --git a/asmrun/spacetime.c b/asmrun/spacetime.c
index bf4b6f3..1f4207b 100644
--- a/asmrun/spacetime.c
+++ b/asmrun/spacetime.c
@@ -27,6 +27,9 @@
 #ifdef HAS_UNISTD
 #include <unistd.h>
 #endif
+#ifdef _WIN32
+#include <process.h> /* for _getpid */
+#endif
 
 #include "caml/alloc.h"
 #include "caml/backtrace_prim.h"
@@ -159,7 +162,7 @@ static void open_snapshot_channel(void)
   char filename[8192];
   int pid;
 #if defined (_WIN32) || defined (_WIN64)
-  pid = Int_val(val_process_id);
+  pid = _getpid();
 #else
   pid = getpid();
 #endif
@@ -188,7 +191,7 @@ static void maybe_reopen_snapshot_channel(void)
 
   int pid;
 #if defined (_WIN32) || defined (_WIN64)
-  pid = Int_val(val_process_id);
+  pid = _getpid();
 #else
   pid = getpid();
 #endif
