diff -u a/Makefile b/Makefile
--- a/Makefile   2015-01-22 13:01:06.000000000 +0000
+++ b/Makefile   2016-03-14 06:42:40.002014399 +0000
@@ -3,17 +3,17 @@
 
 include $(shell cygpath -ad "$(shell ocamlopt -where)/Makefile.config")
 
-MINGW_PREFIX = i686-w64-mingw32
-MINCC = $(MINGW_PREFIX)-gcc
+MINGW_PREFIX = i686-w64-mingw32-
+MINCC = $(MINGW_PREFIX)gcc
 
-MINGW64_PREFIX = x86_64-w64-mingw32
-MIN64CC = $(MINGW64_PREFIX)-gcc
+MINGW64_PREFIX = x86_64-w64-mingw32-
+MIN64CC = $(MINGW64_PREFIX)gcc
 
-CYGWIN_PREFIX = i686-pc-cygwin
-CYGCC = $(CYGWIN_PREFIX)-gcc
+CYGWIN_PREFIX = i686-pc-cygwin-
+CYGCC = $(CYGWIN_PREFIX)gcc
 
-CYGWIN64_PREFIX = x86_64-pc-cygwin
-CYG64CC = $(CYGWIN64_PREFIX)-gcc
+CYGWIN64_PREFIX = x86_64-pc-cygwin-
+CYG64CC = $(CYGWIN64_PREFIX)gcc
 
 .PHONY: version.ml
 version.ml:
diff -u a/reloc.ml b/reloc.ml
--- a/reloc.ml   2015-01-22 16:30:07.000000000 +0000
+++ b/reloc.ml   2016-03-14 06:41:10.041361908 +0000
@@ -1050,8 +1050,8 @@
 
 let setup_toolchain () =
   let mingw_libs pre =
-    gcc := pre ^ "-gcc";
-    objdump := pre ^ "-objdump";
+    gcc := pre ^ "gcc";
+    objdump := pre ^ "objdump";
     search_path :=
       !dirs @
       [
