diff --git a/src/Makefile b/src/Makefile
index b43859b..8125932 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,11 +1,11 @@
 VERSION = 1.10.0
-ifeq "$(shell ocamlc -config |grep os_type)" "os_type: Win32"
+ifeq "$(shell ocamlfind ocamlc -config |grep os_type)" "os_type: Win32"
 EXE=.exe
 else
 EXE=
 endif
 
-NATDYNLINK := $(shell if [ -f `ocamlc -where`/dynlink.cmxa ]; then \
+NATDYNLINK := $(shell if [ -f `ocamlfind ocamlc -where`/dynlink.cmxa ]; then \
                         echo YES; \
                       else \
                         echo NO; \
